openmc/mgxs_library.py

```diff
@@ -379,200 +436,587 @@ def tabular_legendre(self, tabular_legendre):
     @num_polar.setter
     def num_polar(self, num_polar):
         # Make sure we have positive ints
-        check_value("num_polar", num_polar, Integral)
-        check_greater_than("num_polar", num_polar, 0)
+        check_value('num_polar', num_polar, Integral)
+        check_greater_than('num_polar', num_polar, 0)
         self._num_polar = num_polar
 
     @num_azimuthal.setter
     def num_azimuthal(self, num_azimuthal):
-        check_value("num_azimuthal", num_azimuthal, Integral)
-        check_greater_than("num_azimuthal", num_azimuthal, 0)
+        check_value('num_azimuthal', num_azimuthal, Integral)
+        check_greater_than('num_azimuthal', num_azimuthal, 0)
         self._num_azimuthal = num_azimuthal
 
     @total.setter
     def total(self, total):
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
+        """This method sets the total cross section by performing a
+        deep-copy of the provided ndarray. If the angular
+        representation is "isotropic" the shape of the input array
+        must be the number of energy groups. If the angular
+        representation is "angle" then the shape of the input
+        array must be the number of polar angles, number azimuthal
+        angles and energy groups.
+
+        Parameters
+        ----------
+        total: ndarray
+            Array of group-wise cross sections to apply
+
+        """
+
         # check we have a numpy list
-        check_type("total", total, np.ndarray, expected_iter_type=Real)
-        if total.shape == shape:
-            self._total = np.copy(total)
-        else:
-            msg = 'Shape of provided total "{0}" does not match shape ' \
-                  'required, "{1}"'.format(total.shape, shape)
-            raise ValueError(msg)
+        check_type('total', total, np.ndarray, expected_iter_type=Real)
+        # Check the dimensions of the data
```

Comment:
This comment seems a bit redundant, I think it's pretty obvious what the next line is doing (I recognize that a lot of my comments would fall under this same category!)